front/pages/w/[wId]/builder/assistants/index.tsx

```diff
@@ -85,6 +95,18 @@ export default function WorkspaceAssistants({
     }
   })();
 
+  useEffect(() => {
+    const selectedTab = router.query.tabScope;
+    if (
+      router.isReady &&
+      router.route &&
+      typeof selectedTab === "string" &&
+      isValidTab(selectedTab)
+    ) {
+      setActiveTab(selectedTab);
+    }
+  }, [router.route, router.isReady, router.query.tabScope]);
```

Comment:
instead of copying router query to state it might be easier and faster to just read the value from the router .. 
```suggestion
const activeTab = (
      router.isReady &&
      router.route &&
      typeof router.query.tabScope === "string" &&
      isValidTab(router.query.tabScope)
    ) ? router.query.tabScope : "company"
```